include/deal.II/base/quadrature_lib.h

```diff
@@ -647,6 +647,206 @@ class QGaussLobattoChebyshev : public Quadrature<dim>
 
 };
 
+/**
+ * Given an arbitrary quadrature formula, return one that chops the quadrature
+ * points above the hyper-plane defined by $\sum_i x_i = 1$. In other words,
+ * it extracts those quadrature points from the base formula that satisfy
+ * $\sum_i (\mathbf x_q)_i \le 1$."
```

Comment:
The main use of this quadrature formula is not to chop tensor product quadratures. That's of course numerically unstable. The main use of this formula is to enforce that all points are in the reference simplex, so that the resulting quadrature can safely be used to patch arbitrary shapes using simplices, like it is done in `QSplit`. What you pass to this formula of course makes this a sensible quadrature for simplices or a terrible one, just like what you pass to Quadrature(points, weights) is up to you.

If you want to integrate on triangles/tetrahedrons, you should use this class with a quadrature formula that is not a tensor product. In other words, this class is a place-holder for quadrature formulas that only have points in the reference simplex. The fact that the default constructor  chops away points outside the reference simplex doesn't mean that you should use it with a `QGauss` formula and expect good results...

Reply:
This is actually quite a good write-up of the rationale! Can you just copy it into the class doc? I think that would have greatly helped me understand how this class is intended to be used!